node.go

```diff
@@ -602,3 +602,159 @@ func sortFields(fields []Field) {
 func compareFields(a, b Field) int {
 	return strings.Compare(a.Name(), b.Name())
 }
+
+func RewriteSchema(schema *Schema, replace func(path []string, node Node) Node) *Schema {
+	return NewSchema(schema.name, rewriteSchema(nil, schema.root, replace))
+}
+
+func rewriteSchema(path []string, node Node, replace func(path []string, node Node) Node) Node {
+	if node.Leaf() {
+		return replace(path, node)
+	}
+
+	// Types requiring special handling.
```

Comment:
These cases are required for a rewritten SchemaOf(T) to still be functional.  The approach of replacing all non-leaf nodes with `Group` has some issues: